packages/winpanda/extra/src/winpanda/core/utils.py

```diff
@@ -135,45 +120,20 @@ def rc_load_yaml(fpath, emheading=None, render=False, context=None):
     :param context:   ResourceContext, rendering context data object
     :return:          yaml obj, YAML-formatted data
     """
-    assert isinstance(fpath, Path) and fpath.is_absolute(), (
-        f'Argument: fpath: Absolute pathlib.Path is required: {fpath}'
-    )
-
-    if context is None:
-        context_items = {}
-    else:
-        assert isinstance(context, ResourceContext), (
-            f'Argument: context:'
-            f' Got {type(context).__name__} instead of ResourceContext'
+    if render is True:
+        jj2_env = jj2.Environment(
+            loader=jj2.FileSystemLoader(str(fpath.parent))
         )
-        context_items = context.get_items()
-
-    try:
-        if render is True:
-            jj2_env = jj2.Environment(
-                loader=jj2.FileSystemLoader(str(fpath.parent))
-            )
-            jj2_tmpl = jj2_env.get_template(str(fpath.name))
-            yaml_str = jj2_tmpl.render(**context_items)
-            LOG.debug(f'rc_load_yaml(): yaml_str: {yaml_str}')
-            y_body = yaml.safe_load(yaml_str)
-        else:
-            with fpath.open() as fp:
-                y_body = yaml.safe_load(fp)
-    except (FileNotFoundError, jj2.TemplateNotFound) as e:
-        err_msg = f'Load: {fpath}'
-        err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
-        raise cr_exc.RCNotFoundError(err_msg) from e
-    except (OSError, RuntimeError) as e:
-        err_msg = f'Load: {fpath}: {type(e).__name__}: {e}'
-        err_msg = f'{emheading}: {err_msg}' if emheading else err_msg
-        raise cr_exc.RCError(err_msg) from e
-    except (jj2.TemplateError, yaml.YAMLError) as e:
```

Comment:
`yaml.YAMLError` is not caught by the decorator.

Reply:
Tests added, exception handler added in decorator